include/material/bingham.tcc

```diff
@@ -0,0 +1,107 @@
+//! Read material properties
+template <unsigned Tdim>
+void mpm::Bingham<Tdim>::properties(const Json& material_properties) {
+  try {
+    density_ = material_properties["density"].template get<double>();
+    youngs_modulus_ =
+        material_properties["youngs_modulus"].template get<double>();
+    poisson_ratio_ =
+        material_properties["poisson_ratio"].template get<double>();
+    tau0_ = material_properties["tau0"].template get<double>();
+    mu_ = material_properties["mu"].template get<double>();
+    critical_shear_rate_ =
+        material_properties["critical_shear_rate"].template get<double>();
+    properties_ = material_properties;
+    status_ = true;
+  } catch (std::exception& except) {
+    std::cerr << "Material parameter not set: " << except.what() << '\n';
+  }
+}
+
+//! Elastic tensor is not defined in Bingham model, throws an error
+template <unsigned Tdim>
+Eigen::Matrix<double, 6, 6> mpm::Bingham<Tdim>::elastic_tensor() {
+
+  Eigen::Matrix<double, 6, 6> de;
+  de.setZero();
+
+  throw std::runtime_error("Elastic tensor is not used for this material");
+
+  return de;
+}
+
+//! Compute stress without a particle handle is undefined in the Bingham model,
+//! throws an error
+template <unsigned Tdim>
+Eigen::Matrix<double, 6, 1> mpm::Bingham<Tdim>::compute_stress(
+    const Vector6d& stress, const Vector6d& dstrain) {
+
+  Vector6d stress_results;
+  stress_results.setZero();
+
+  throw std::runtime_error("Stress computation for this material is not valid");
+
+  return stress_results;
+}
+
+//! Compute stress
+template <unsigned Tdim>
+Eigen::Matrix<double, 6, 1> mpm::Bingham<Tdim>::compute_stress(
+    const Vector6d& stress, const Vector6d& dstrain,
+    const ParticleBase<Tdim>* ptr) {
+
+  const unsigned phase = 0;
+  const auto strain_rate = ptr->strain_rate(phase);
```

Comment:
`strain_rate` is not the same as `D`, please see the documentation.

Reply:
Understood. Thanks for this.